src/cargo/core/compiler/timings.js

```diff
@@ -23,6 +23,36 @@ let UNIT_COORDS = {};
 let REVERSE_UNIT_DEPS = {};
 let REVERSE_UNIT_RMETA_DEPS = {};
 
+function scale_to_graph_width(scale) {
+    // Cap the size of the graph. It is hard to view if it is too large, and
+    // browsers may not render a large graph because it takes too much memory.
+    // 4096 is still ridiculously large, and probably won't render on mobile
+    // browsers, but should be ok for many desktop environments.
+    return Math.min(200 + scale * 20, 4096);
+}
+
+// Guess a reasonable initial scale value to fit the graphs to the width of the user's window.
+function init_graph_scale() {
+    const width = window.innerWidth;
+    // Leave some margin around the chart, to account for axis labels
+    const max_available_space = Math.max(200, width - 150);
+
+    for (let scale = 1; scale < 100; scale++) {
```

Comment:
Can't we calculate this rather than do a linear search?

Reply:
Yeah, we can. But the reason I didn't do that is because we would have to implement the scaling logic twice, because it's not "reversible". We have one function that computes the width based on the scale, and in the other direction we'd need a function that computes the scale from the width. So we'd have to either duplicate the logic or create some shared constants on which those two functions would operate, but both risks the logic diverging. The for loop makes sure that we can't have a bug in the mapping between those two directions, because we only implement the direction scale -> width.

But if you don't like it, I can just reimplement the mapping logic from the other direction.